### PR TITLE
[4.0] Uninstalled components with leftover menu records appear in the Components Dashboard

### PR DESCRIPTION
**Pull request by nikosdion**
### Summary of Changes

Fixes the Components Dashboard to work equivalently to the backend Main Menu with regards to uninstalled / disabled components.

### Testing Instructions

* Install a component, ANY component, e.g. JCE, com_weblinks, whatever.
* Delete the `#__extensions` record of the component to simulate a botched component uninstallation which leaves the component menu items behind

### Actual result BEFORE applying this Pull Request

The main menu (sidebar) does not display links to the component.

However, the Components Dashboard (four squares icon next to the “Components” menu item name in the main menu) displays the missing component's links, untranslated. Clicking on them results in a 404 Component Not Found error page as expected.

### Expected result AFTER applying this Pull Request

The main menu (sidebar) does not display links to the component.

The Components Dashboard (four squares icon next to the “Components” menu item name in the main menu) does not display links to the component.

### Documentation Changes Required

None.

### Further information

This problem was observed on a live site a while after after we uninstalled a (paid) extension. Due to an internal error in the extension's script the uninstallation didn't complete. Unbeknownst to us at the time, the extension record was removed from the `#__extensions` table, the files were removed from the filesystem **BUT** the `#__menu` records were left behind.

We didn't think much of this uninstallation issue until a few weeks later when we started doing the final, detailed inspection of the site to get it ready for delivery. @crystalenka noticed that the Components Dashboard would display links to the extension whereas the sidebar Main Menu wouldn't. I took a 5' look and found the discrepancy between how mod_submenu and mod_menu post-process the loaded menu items. This PR addresses this discrepancy.

No, the offending component was NOT mine, NEITHER was it JCE or WebLinks. :trollface: In any case the problem is not extension-specific, it can happen if the uninstallation is interrupted before the `#__menu` entries are removed for any reason.

Ideally, the common post-processing logic should be moved outside of `Joomla\Module\Submenu\Administrator\Menu\Menu` and `Joomla\Module\Menu\Administrator\Menu\CssMenu` and into `Joomla\Component\Menus\Administrator\Helper\MenusHelper` so it can be reused by third party menu modules.

Please note that this is supposed to be my week off. I may not be very responsive until January 3rd when I'm back at the office. I don't even know how I got suckered into fixing this in my week off. Something something something can't say no to my wife and Joomla being Hotel California... Anyway, Merry Christmas / Happy Holidays and Happy New Year, y'all!

### CODE REVIEW

**Comment by toivo** [comment]
I have tested this item :white_check_mark: successfully on 27bbd5b6128bb61383d2d11452570feab89ed9c7<br /><br />Tested successfully in 4.06-dev using PHP 8.0.13<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36441">issues.joomla.org/tracker/joomla-cms/36441</a>.</sub>

**Comment by crystalenka** [comment]
I have tested this item :white_check_mark: successfully on 27bbd5b6128bb61383d2d11452570feab89ed9c7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36441">issues.joomla.org/tracker/joomla-cms/36441</a>.</sub>

File: administrator/modules/mod_submenu/src/Menu/Menu.php [hunk 1]
```diff
@@ -52,6 +53,29 @@ public static function preprocess($parent)
...
 		{
+			if (substr($item->link, 0, 8) === 'special:')
+			{
+				$special = substr($item->link, 8);
+
+				if ($special === 'language-forum')
+				{
+					$item->link = 'index.php?option=com_admin&amp;view=help&amp;layout=langforum';
+				}
+			}
```
Comment (richard67) [inline administrator/modules/mod_submenu/src/Menu/Menu.php:64]: @nikosdion Is this also related to the issue, or is this something else?

Comment (nikosdion) [reply administrator/modules/mod_submenu/src/Menu/Menu.php:64]: Copied over from the mani menu module. Shouldn't be needed, but I'm pretty sure someone thought the same about the `element` property not being defined in the past. So, in an effort to minimise future surprises, I copied that code over.

Comment (brianteeman) [reply administrator/modules/mod_submenu/src/Menu/Menu.php:64]: so why didnt you copy over the custom-forum as well?

Comment (nikosdion) [reply administrator/modules/mod_submenu/src/Menu/Menu.php:64]: I thought it would be obvious: because that relies on configuration options in the main menu module which do not exist in the submenu module.

Don't get me started on Joomla's inconsistency.

Comment (brianteeman) [reply administrator/modules/mod_submenu/src/Menu/Menu.php:64]: oh - sorry I thought they were both configured in the options

Comment (nikosdion) [reply administrator/modules/mod_submenu/src/Menu/Menu.php:64]: Yup, I originally had thought the same and was quite surprised it's not the case. That's why I didn't implement my suggestion of having common code for the menu items pre–processing. It requires some discussion about moving options which might be a something that makes sense only in a new minor or major release, not a point release.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36441">issues.joomla.org/tracker/joomla-cms/36441</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!
